### PR TITLE
Add comprehensive unit tests to cover edge cases for Delta log path parsing

### PR DESCRIPTION
**Pull request by hackintoshrao**
This PR introduces new unit tests to validate the Delta Lake log path parsing logic corner cases thoroughly:

1. test_invalid_version_and_part_lengths:
   - Verifies correct handling of invalid version lengths (VERSION_LEN)
   - Tests parsing of invalid multipart checkpoint lengths (MULTIPART_PART_LEN)
   - Ensures proper error handling for invalid UUID lengths (UUID_PART_LEN)

2. test_empty_filename:
   - Confirms that URLs ending with a slash (resulting in empty filenames)
     are correctly rejected

3. test_ok_none_and_unknown_cases:
   - Validates handling of non-numeric version prefixes (Ok(None) cases)
   - Verifies correct parsing of missing file extensions (Ok(None) cases)
   - Ensures proper handling of empty file extensions (Ok(Some) with Unknown type)

These tests significantly improve coverage of edge cases and error
conditions in log path parsing.

### CODE REVIEW

File: kernel/src/path.rs [hunk 1]
```diff
@@ -199,6 +199,88 @@ mod tests {
...
     #[test]
+    fn test_invalid_version_and_part_lengths() {
+        let table_log_dir = table_log_dir_url();
+
+        // Test cases for VERSION_LEN (20 digits)
+        let test_cases = vec![
+            // Error expected: Version number is 19 digits, but should be 20
+            ("000000000000000006.json", "short version number"),
+            // Error expected: Version number is 21 digits, but should be 20
+            ("000000000000000000006.json", "long version number"),
```
Comment (scovich) [inline kernel/src/path.rs:210]: These two cases are already covered?
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L227-L233

File: kernel/src/path.rs [hunk 2]
```diff
@@ -199,6 +199,88 @@ mod tests {
...
+                "Expected InvalidLogPath error for {}",
+                error_msg
+            );
+        }
+
+        // Test cases for MULTIPART_PART_LEN (10 digits)
+        let test_cases = vec![
+            // Error expected: Part number is 9 digits, but should be 10
+            (
+                "00000000000000000010.checkpoint.000000001.0000000002.parquet",
```
Comment (scovich) [inline kernel/src/path.rs:228]: Already covered?
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L449-L452

File: kernel/src/path.rs [hunk 3]
```diff
@@ -199,6 +199,88 @@ mod tests {
...
+                "short part number",
+            ),
+            // Error expected: Part number is 11 digits, but should be 10
+            (
+                "00000000000000000010.checkpoint.00000000001.0000000002.parquet",
+                "long part number",
+            ),
+            // Error expected: Total parts is 9 digits, but should be 10
+            (
+                "00000000000000000010.checkpoint.0000000001.000000002.parquet",
```
Comment (scovich) [inline kernel/src/path.rs:238]: Already covered?
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L454-L457

File: kernel/src/path.rs [hunk 4]
```diff
@@ -199,6 +199,88 @@ mod tests {
...
+                "Expected InvalidLogPath error for {}",
+                error_msg
+            );
+        }
+
+        // Test cases for UUID_PART_LEN (36 characters)
+        let test_cases = vec![
+            // Error expected: UUID is 35 characters, but should be 36
+            (
+                "00000000000000000010.checkpoint.3a0d65cd-4056-49b8-937b-95f9e3ee90.parquet",
```
Comment (scovich) [inline kernel/src/path.rs:263]: Technically already covered:
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L363-L366

... but this is nicer (checks the boundary case)

File: kernel/src/path.rs [hunk 5]
```diff
@@ -518,4 +600,58 @@ mod tests {
             .unwrap();
         ParsedLogPath::try_from(log_path).expect_err("non-numeric hi");
     }
+
+    #[test]
+    fn test_empty_filename() {
+        let table_log_dir = table_log_dir_url();
+
+        // Test URL ending with a slash
```
Comment (scovich) [inline kernel/src/path.rs:608]: Already covered?
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L205-L210

File: kernel/src/path.rs [hunk 6]
```diff
@@ -518,4 +600,58 @@ mod tests {
...
+            "Expected InvalidLogPath error for empty filename"
+        );
+    }
+
+    #[test]
+    fn test_ok_none_and_unknown_cases() {
+        let table_log_dir = table_log_dir_url();
+
+        // Test cases for non-numeric version (should return Ok(None))
+        let test_cases = vec!["_last_checkpoint", "abc.json", "version_001.json"];
```
Comment (scovich) [inline kernel/src/path.rs:623]: 2/3 cases already covered?
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L205-L210
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L222-L225

File: kernel/src/path.rs [hunk 7]
```diff
@@ -518,4 +600,58 @@ mod tests {
...
+            let result = ParsedLogPath::try_from(log_path);
+            assert!(
+                matches!(result, Ok(None)),
+                "Expected Ok(None) for non-numeric version: {}",
+                file_name
+            );
+        }
+
+        // Test case for missing file extension (should return Ok(None))
+        let log_path = table_log_dir.join("00000000000000000010").unwrap();
```
Comment (scovich) [inline kernel/src/path.rs:636]: Already covered?
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L217-L220

File: kernel/src/path.rs [hunk 8]
```diff
@@ -518,4 +600,58 @@ mod tests {
...
+        let result = ParsedLogPath::try_from(log_path);
+        assert!(
+            matches!(result, Ok(None)),
+            "Expected Ok(None) for missing file extension"
+        );
+
+        // Test case for empty extension (should return Ok(Some) with Unknown file type)
+        let log_path = table_log_dir.join("00000000000000000011.").unwrap();
```
Comment (scovich) [inline kernel/src/path.rs:644]: Nice catch. I didn't even think of this case when first writing these tests.

File: kernel/src/path.rs (hunk 4 above)
Comment (scovich) [reply kernel/src/path.rs:263]: It's off-by-one tho (only 34 chars):
```suggestion
            // Error expected: UUID is 35 characters, but should be 36
            (
                "00000000000000000010.checkpoint.3a0d65cd-4056-49b8-937b-95f9e3ee90e.parquet",
```

File: kernel/src/path.rs (hunk 7 above)
Comment (hackintoshrao) [reply kernel/src/path.rs]: Removed.

File: kernel/src/path.rs (hunk 8 above)
Comment (hackintoshrao) [reply kernel/src/path.rs]: Retained this.

File: kernel/src/path.rs (hunk 5 above)
Comment (hackintoshrao) [reply kernel/src/path.rs]: Removed.

File: kernel/src/path.rs (hunk 4 above)
Comment (hackintoshrao) [reply kernel/src/path.rs]: Fixed char length and retained the case.
